test/jdk/java/awt/Robot/ScreenCaptureRobotTest.java

```diff
@@ -0,0 +1,147 @@
+    private static Frame frame;
+    private static volatile Canvas canvas;
+    private static BufferedImage realImage;
+    private static BufferedImage displayImage;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::initializeGUI);
+            doTest();
+        } finally {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::disposeFrame);
+        }
+    }
+
+    private static void initializeGUI() {
+        frame = new Frame("ScreenCaptureRobotTest Frame");
+        realImage = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice().getDefaultConfiguration()
+                .createCompatibleImage(200, 100);
+
+        Graphics g = realImage.createGraphics();
+        g.setColor(Color.yellow);
+        g.fillRect(0, 0, 200, 100);
+        g.setColor(Color.red);
+        g.setFont(new Font("SansSerif", Font.BOLD, 20));
+        g.drawString("Capture This", 10, 40);
+        g.dispose();
+        displayImage = realImage;
+
+        canvas = new ImageCanvas();
+        canvas.setBackground(Color.YELLOW);
+        frame.setLayout(new BorderLayout());
+        frame.add(canvas);
+        frame.setSize(300, 200);
+        frame.setLocation(100, 100);
+        frame.setVisible(true);
+        canvas.requestFocus();
+    }
+
+    private static void doTest() throws Exception {
+        Robot robot;
+        robot = new Robot();
+        robot.delay(delay);
+        robot.waitForIdle();
+
+        Point pnt = canvas.getLocationOnScreen();
+        Rectangle rect = new Rectangle(pnt.x + 10, pnt.y + 10, 200, 100);
+
+        // Capturing Image using Robot
+        BufferedImage capturedImage = robot.createScreenCapture(rect);
+
+        if (!compareImages(capturedImage, realImage)) {
+            String errorMessage = "FAIL : Captured Image is different from "
+                    + "the actual image";
+            System.err.println("Test failed");
+            throw new RuntimeException(errorMessage);
+        } else {
+            System.out.println("\nCaptured Image is same as actual Image");
+            System.out.println("Test passed");
+        }
+    }
+
+    private static boolean compareImages(BufferedImage capturedImg,
+            BufferedImage realImg) {
+        int capturedPixel;
+        int realPixel;
+        int imgWidth;
+        int imgHeight;
+        boolean result = true;
+
+        imgWidth = capturedImg.getWidth(null);
+        imgHeight = capturedImg.getHeight(null);
+
+        for (int i = 0; i < (imgWidth); i++) {
+            for (int j = 0; j < (imgHeight); j++) {
+                capturedPixel = capturedImg.getRGB(i, j);
+                realPixel = realImg.getRGB(i, j);
+                if (capturedPixel != realPixel) {
+                    result = false;
+                    return result;
+                }
+            }
+        }
+        return result;
```

Comment:
```suggestion
        return true;
```

Reply:
In fact, image comparison can be borrowed from the `Util` class (in Swing part):

https://github.com/openjdk/jdk/blob/1f3574855e79221739d8800235583b7c47ebae97/test/jdk/javax/swing/regtesthelpers/Util.java#L59-L63

Reply:
I agree to @aivanov-jdk .

Reply:
Used the same logic as in `Util` class. 
Added image width & height check now.

Reply:
I meant _using `Util.compareBufferedImages`_ rather than copying the code.

Reply:
@aivanov-jdk 
Definitely using Util will be simpler and lesser lines of code.
Then we will be getting a True/False as return value.
Afraid that we will have less info on pixel, color & coordinates like
`Captured pixel (fffeff00) at (0, 0) is not equal to real pixel (ffffff00)`

This being an AWT Robot test, should we add a Swing Util dependency as well?

Reply:
The point of this test seems to be a Basic-Acceptance-Test for Robot screen capture.
As such I think it should NOT borrow code. It should be self-contained.
And with such a purpose should work 100% of the time - so long as colour profiles are set up right.

Reply:
> The point of this test seems to be a Basic-Acceptance-Test for Robot screen capture. As such I think it should NOT borrow code. It should be self-contained.

Agreed.

> And with such a purpose should work 100% of the time - so long as colour profiles are set up right.

So far the test isn't stable enough.

Reply:
May be, Is there any possibility of adding some extra Stability checks,
to handle local failures.